test/microbench/fam-api-mb/fam_region_spanning.cpp

```diff
@@ -0,0 +1,580 @@
+    do {
+        dataitemName = getRandomString() ;
+        hashVal = hash<string>{}(dataitemName) % (msrvcnt);
+        for (int i = 0; i< mcnt; i++)
+                if (hashVal == (size_t)memsrv[i])
+                    {found = 1; break;}
+    } while (!found);
+    return dataitemName;
+}
+
+string getDataitemName(int peId, int totalPEs) {
+    size_t hashVal;
+    size_t val = 0;
+    string dataitemName;
+    int switch_cnt = 8;
+    switch(msrvcnt) {
+        case 1:         val = peId;
+                        break;
+        case 2 :        val = peId/switch_cnt;
+                        break;
+        case 4 :
+                        if ( peId < switch_cnt)
+                                val =  peId % 2;
+                        else
+                                val = msrvcnt / 2 + peId % 2;
+                        break;
+        case 8 :
+                        if ( peId < switch_cnt)
+                                val = peId/switch_cnt + peId % 4;
+                        else
+                                val = msrvcnt / 2 + peId % 4;
+                        break;
+        case 16:
+                        val = peId;
+                        break;
+
+    }
+
+    do {
+        dataitemName = getRandomString() ;
+        hashVal = hash<string>{}(dataitemName) % totalPEs;
+    } while (hashVal != (size_t)val);
+    return dataitemName;
+}
+
+TEST(FamRegionSpanning, FamGetSingleRegionForAllPEs) {
+    int64_t *local = (int64_t *)malloc(gDataSize);
+    uint64_t offset = 0;
+    warmup_function(fam_get_blocking,local, itemLocal[i], offset, gDataSize);
+    EXPECT_NO_THROW(my_fam->fam_barrier_all());
+
+    for (int i = 0; i < NUM_MM_ITERATIONS; i++) {
+        for (int io =0; io < NUM_IO_ITERATIONS; io++) {
+
+                EXPECT_NO_THROW(my_fam->fam_get_blocking(local, itemLocal[i], offset, gDataSize));
+        }
+
+    }
+    EXPECT_NO_THROW(my_fam->fam_barrier_all());
+
+    free(local);
+}
+
+TEST(FamRegionSpanning, FamPutSingleRegionForAllPEs) {
+    int64_t *local = (int64_t *)malloc(gDataSize);
+    uint64_t offset = 0;
+    warmup_function(fam_put_blocking,local, itemLocal[i], offset, gDataSize);
+    EXPECT_NO_THROW(my_fam->fam_barrier_all());
+
+    for (int i = 0; i < NUM_MM_ITERATIONS; i++) {
+        for (int io =0; io < NUM_IO_ITERATIONS; io++) {
+
+                EXPECT_NO_THROW(my_fam->fam_put_blocking(local, itemLocal[i], offset, gDataSize));
+        }
+
+    }
+    EXPECT_NO_THROW(my_fam->fam_barrier_all());
+
+    free(local);
+}
+
+
+// Test case -  FamNonBlockingPut test.
+TEST(FamRegionSpanning, FamNonBlockingPutSingleRegionForAllPEs) {
+    int64_t *local = (int64_t *)malloc(gDataSize);
+    uint64_t offset = 0;
+    warmup_function(fam_put_blocking,local, itemLocal[i], offset, gDataSize);
+    EXPECT_NO_THROW(my_fam->fam_barrier_all());
+
+    for (int i = 0; i < NUM_MM_ITERATIONS; i++) {
+        for (int io =0; io < NUM_IO_ITERATIONS; io++) {
+/*               if ( io % 10 ==  0)
```

Comment:
For non blocking i believe the loops should be other way around - For each IO iteration write to all the data items. Thoughts? @faizan-barmawer